src/algebra/continued_fractions/convergents_equiv.lean

```diff
@@ -348,40 +347,32 @@ begin
-            obtain ⟨gp_succ_m, s_succ_mth_eq⟩ : ∃ gp_succ_m, g.s.nth (m + 1) = some gp_succ_m, from
-              option.ne_none_iff_exists'.mp not_terminated_at_n,
-            obtain ⟨gp_m, mth_s_eq⟩ : ∃ gp_m, g.s.nth m = some gp_m, from
-              g.s.ge_stable m.le_succ s_succ_mth_eq,
-            -- we then plug them into the recurrence
-            suffices : 0 < gp_m.a ∧ 0 < gp_m.b + gp_succ_m.a / gp_succ_m.b, by {
-              have : g'.s.nth m = some ⟨gp_m.a, gp_m.b + gp_succ_m.a / gp_succ_m.b⟩, from
-                squash_seq_nth_of_not_terminated mth_s_eq s_succ_mth_eq,
-              have : gp' = ⟨gp_m.a, gp_m.b + gp_succ_m.a / gp_succ_m.b⟩, by cc,
-              rwa this },
-            split,
-            { exact (s_pos (nat.lt.step m_lt_n) mth_s_eq).left },
-            { have : 0 < gp_m.b, from (s_pos (nat.lt.step m_lt_n) mth_s_eq).right,
-              have : 0 < gp_succ_m.a / gp_succ_m.b, by
-              { have : 0 < gp_succ_m.a ∧ 0 < gp_succ_m.b, from
-                  s_pos (lt_add_one $ m + 1) s_succ_mth_eq,
-                exact (div_pos this.left this.right) },
-              linarith } },
-          { -- the easy case: before the squashed position, nothing changes
-            have : g.s.nth m = some gp', by {
-              have : g'.s.nth m = g.s.nth m, from squash_gcf_nth_of_lt succ_m_lt_n,
-              rwa this at s_mth_eq' },
-            exact s_pos (nat.lt.step $ nat.lt.step succ_m_lt_n) this } },
-        rwa [(IH this).symm] },
+      { rwa ← IH,
+        assume gp' m m_lt_n s_mth_eq',
+        -- case distinction on m + 1 = n or m + 1 < n
+        cases m_lt_n with n succ_m_lt_n,
+        { -- the difficult case at the squashed position: we first obtain the values from
+          -- the sequence
+          obtain ⟨gp_succ_m, s_succ_mth_eq⟩ : ∃ gp_succ_m, g.s.nth (m + 1) = some gp_succ_m, from
+            option.ne_none_iff_exists'.mp not_terminated_at_n,
+          obtain ⟨gp_m, mth_s_eq⟩ : ∃ gp_m, g.s.nth m = some gp_m, from
+            g.s.ge_stable m.le_succ s_succ_mth_eq,
+          -- we then plug them into the recurrence
+          suffices : 0 < gp_m.a ∧ 0 < gp_m.b + gp_succ_m.a / gp_succ_m.b, by {
+            have ot : g'.s.nth m = some ⟨gp_m.a, gp_m.b + gp_succ_m.a / gp_succ_m.b⟩, from
+              squash_seq_nth_of_not_terminated mth_s_eq s_succ_mth_eq,
+            have : gp' = ⟨gp_m.a, gp_m.b + gp_succ_m.a / gp_succ_m.b⟩, by cc,
+            rwa this },
+          refine ⟨(s_pos (nat.lt.step m_lt_n) mth_s_eq).left, _⟩,
+          refine add_pos (s_pos (nat.lt.step m_lt_n) mth_s_eq).right _,
+          have : 0 < gp_succ_m.a ∧ 0 < gp_succ_m.b := s_pos (lt_add_one $ m + 1) s_succ_mth_eq,
+          exact (div_pos this.left this.right) },
+        { -- the easy case: before the squashed position, nothing changes
+          refine s_pos (nat.lt.step $ nat.lt.step succ_m_lt_n) _,
+          exact eq.trans (squash_gcf_nth_of_lt succ_m_lt_n).symm s_mth_eq' } },
```

Comment:
It looks like the original author of this proof was trying really hard to make it readable without access to a tactic state. I'll let another reviewer decide whether we care about losing some of that, but I suspect the answer is that we don't really, meaning this change is fine.

Reply:
Ok, I found that the proof "flows" better with the change, but I can see that it is slightly less humanly readable.  I also have the impression that, with the change, it is faster, but I have not actually measured this.